src/frontend/ui/tool/mod.rs

```diff
@@ -0,0 +1,30 @@
+use crate::frontend::{rectangle::Rectangle, screenshot_window::MouseEvent};
+
+use super::drawable::Drawable;
+
+pub mod crop;
+
+#[derive(Debug, PartialEq, Eq, Hash)]
+pub enum ToolIdentifier {
+    Crop,
+}
+
+pub enum ToolCommand {
+    Noop,
+    Crop(Rectangle),
+}
+
+pub trait Tool {
```

Comment:
What about enforcing a boxed function, since the tool-manager requires the tools to be boxed?
```rust
    fn boxed() -> Box<Self>;
```

Reply:
That's not a function related to a tool. Also we might run into the case where a specific tool has to be constructed in a different way. So I don't want to add it to the Tool trait.

Reply:
ok